_stbt/control.py

```diff
@@ -263,6 +265,107 @@ def _connect():
     return LircRemote(control_name, _connect)
 
 
+class RemoteFrameBuffer(object):
+    """Send a key-press to a set-top box running a VNC Remote Frame Buffer
+        protocol.
+        Expected key press input: 
+            <KEY_LABEL>(<KEY_HEX_CODE>)
+ 
+        control = RemoteFrameBuffer("192.168.0.123")
+        control.press("MENU(0xE001)")
+    """
+    def __init__(self, hostname, port=None):
+        self.hostname = hostname
+        self.port = int(port or 5900)
+        self.fail_on_error = True
```

Comment:
What's the motivation for `fail_on_error`?  Raising an exception if a keypress fails is the expected behaviour for the other controls and removing this would greatly simplify your implementation.  We could potentially strip out all the explicit error handling if `fail_on_error` were to be removed.